File: examples/src/main/scala/sample/java/parameterizedtype/generic/ArrayStack.java
package sample.java.parameterizedtype.generic;

import java.util.*;
class ArrayStack<E> implements Stack<E> {
    private List<E> list;
    public ArrayStack() { list = new ArrayList<E>(); }
    public boolean empty() { return list.size() == 0; }
    public void push(E elt) { list.add(elt); }
    public E pop() {
      E elt = list.remove(list.size()-1);
      return elt;
    }
    public String toString() { return "stack"+list.toString(); }
}

File: examples/src/main/scala/sample/scala/parameterizedtype/generic/ArrayStack.scala
package sample.scala.parameterizedtype.generic

import scala.collection.mutable.ArrayBuffer
class ArrayStack[E] extends Stack[E]{
  private val list:ArrayBuffer[E] = new ArrayBuffer[E]()
  
  def empty():Boolean = {
    return list.size == 0
  }
  
  def push(elt:E):Unit = {
    list += elt
  }
  def pop():E = {
    val elt:E = list.remove(list.size-1)
    return elt
  }
  
  override def toString():String = {
    return "stack"+list.toString.drop(11)
  }
}